### PR TITLE
Update iOS SDK version for Kotlin Multiplatform SDK 0.4.0 release

### PR DESCRIPTION
**Pull request by 0xnm**
This PR updates iOS SDK version for Kotlin Multiplatform SDK 0.4.0 release.

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/logs/log_collection/kotlin-multiplatform.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ kotlin {
     | 0.0.1                                    | 2.14.1                  |
     | 0.0.2                                    | 2.17.0                  |
     | 0.0.3                                    | 2.17.0                  |
+    | 0.4.0                                    | 2.20.0                  |
```
Comment (neko-dd) [inline content/en/logs/log_collection/kotlin-multiplatform.md:58]: Should this be `0.0.4`, which more closely matches the ones above it?

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/setup/kotlin-multiplatform.md [hunk 2]
```diff
@@ -71,6 +71,7 @@ Add the following Datadog iOS SDK dependencies, which are needed for the linking
 | 0.0.1                                    | 2.14.1                  |
 | 0.0.2                                    | 2.17.0                  |
 | 0.0.3                                    | 2.17.0                  |
+| 0.4.0                                    | 2.20.0                  |
```
Comment (neko-dd) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/setup/kotlin-multiplatform.md:74]: Ditto above

File: content/en/logs/log_collection/kotlin-multiplatform.md (hunk 1 above)
Comment (0xnm) [reply content/en/logs/log_collection/kotlin-multiplatform.md:58]: no, this is a correct change, see https://github.com/DataDog/dd-sdk-kotlin-multiplatform/tags

Comment (neko-dd) [reply content/en/logs/log_collection/kotlin-multiplatform.md:58]: Great, just wanted to double-check!

**Review by neko-dd — APPROVED** [review]

**Pull request merged**